### PR TITLE
env-vars: Add ODMDATA config vars for nvidia boards

### PR DESCRIPTION
**Pull request by thgreasi**
Change-type: minor
HQ: https://github.com/balena-io/balena/issues/2166
See: https://github.com/balena-io/balenacloud-dashboard/issues/3413
Depends-on: https://github.com/balena-io/balena-supervisor/issues/1206
See: https://www.flowdock.com/app/rulemotion/r-supervisor/threads/54GM3Zqzk75loM4ELl77Zt1w7Vt
Depends-on: https://github.com/balena-io/balena-supervisor/issues/1206
See: https://codepen.io/thgreasi/pen/poyzMRz
Signed-off-by: Thodoris Greasidis <thodoris@balena.io>

### CODE REVIEW

File: src/lib/env-vars.ts [hunk 1]
```diff
@@ -153,6 +153,33 @@ export const DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES: Array<{
...
+	{
+		capableDeviceTypes: [
+			'blackboard-tx2',
+			'jetson-tx2',
+			'n510-tx2',
+			'orbitty-tx2',
+			'skx2',
+			'spacely-tx2',
+			'srd3-tx2',
+		],
```
Comment (20k-ultra) [inline src/lib/env-vars.ts:165]:
```suggestion
		capableDeviceTypes: [
			'astro-tx2',
			'blackboard-tx2',
			'jetson-tx2',
			'n310-tx2',
			'n510-tx2',
			'orbitty-tx2',
			'spacely-tx2',
			'srd3-tx2',
		],
```

Updated list from @acostach.

File: src/features/vars-schema/env-vars.ts [hunk 2]
```diff
@@ -153,6 +153,34 @@ export const DEVICE_TYPE_SPECIFIC_CONFIG_VAR_PROPERTIES: Array<{
...
+				oneOf: [
+					{ const: 1, title: 'Configuration #1' },
+					{ const: 2, title: 'Configuration #2' },
+					{ const: 3, title: 'Configuration #3' },
+					{ const: 4, title: 'Configuration #4' },
+					{ const: 5, title: 'Configuration #5' },
+					{ const: 6, title: 'Configuration #6' },
+				],
+				description:
+					'Define the ODMDATA configuration. Only supported by supervisor versions >= v10.???.???.',
```
Comment (20k-ultra) [inline src/features/vars-schema/env-vars.ts:179]:
```suggestion
					'Define the ODMDATA configuration. Only supported by Supervisor versions >= v11.13.0.',
```

**Review by 20k-ultra — CHANGES_REQUESTED** [review]

File: src/features/vars-schema/env-vars.ts (hunk 2 above)
Comment (thgreasi) [reply src/features/vars-schema/env-vars.ts:179]: I left the `supervisor` in lower case, for consistency with all other occurrences of it.

**Review by 20k-ultra — APPROVED** [review]
Very elegant how this adds a new configurable input. Nice work!

**Comment by thgreasi** [comment]
I'm glad that you liked it @20k-ultra . I will give this an extra test on my BoB tomorrow (since this has been rebased for a while now) and get this merged.

**Pull request merged**